### PR TITLE
Adding an indicator for opening soon. #74

### PR DESCRIPTION
**Pull request by sbeins55**
### What
_Brief description of changes._
This change introduces an indicator for locations that are opening soon. Two labels have been added to the card. One for Distributing and one for receiving.

image

### Why
_What problem does this pull requests solve?  Reference related issues if appropriate._
Issue #74 calls for adding an indicator for locations that are "opening soon"

### Ensure the following interactions work as expected. Please test using a mobile form factor.
- [x] Tapping on a marker on the map displays information about the marker in a popup.
- [x] Tapping the "Show list of locations" button replaces the map view with a list view.
- [x] Tapping an item in the list replaces the list view with a map view, and navigates the map to the tapped item on the map.
- [x] Tapping one of the ✅'s in the legend filters items on the map and in the list.

### Check the app in the following web browsers:
- [x] Chrome
- [x] Firefox
- [ ] Edge
- [x] Safari

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -131,6 +131,21 @@ const createListItem = (location, status, lng, lat) => {
   if (location.seekingVolunteers && location.seekingVolunteers.match(/(?:\byes\b)/i)) {
     seekingVolunteers = `<span class="seekingVolunteers location-list--badge">Needs Volunteer Support</span>`
   }
+
+  const openTimeDistribution = moment(location.openingForDistributingDontations, ["h:mm A"])
+  const openTimeDistributionLessOne = moment(location.openingForDistributingDontations, ["h:mm A"]).subtract(1, 'hours')
+  let openingSoonForDistribution = ''
+  if (moment().isBetween(openTimeDistributionLessOne, openTimeDistribution)) {
+    openingSoonForDistribution = `<p class="opening-soon">Opening soon! ${openTimeDistribution.format("LT")} for distributing</p>`
```
Comment (alexives) [inline script.js:139]: We'll need to tag these classes for translation -- @caseyhelbling or @andyinabox may have some thoughts on how to do this.

```suggestion
    openingSoonForDistribution = `<p class="opening-soon"><span data-translation-id="opening_soon">Opening soon!</span> ${openTimeDistribution.format("LT")} <span data-translation-id="for_distribution">for distributing</span></p>`
```

File: script.js [hunk 2]
```diff
@@ -131,6 +131,21 @@ const createListItem = (location, status, lng, lat) => {
...
+  }
+
+  const openTimeReceiving = moment(location.openingForReceivingDontations, ["h:mm A"])
+  const openTimeReceivingLessOne = moment(location.openingForReceivingDontations, ["h:mm A"]).subtract(1, 'hours')
+  let openingSoonForReceiving = ''
+  if (moment().isBetween(openTimeReceivingLessOne, openTimeReceiving)) {
+    openingSoonForReceiving = `<p class="opening-soon">Opening soon! ${openTimeReceiving.format("LT")} for receiving</p>`
```
Comment (alexives) [inline script.js:146]: Same translation comment...
```suggestion
    openingSoonForReceiving = `<p class="opening-soon"><span data-translation-id="opening_soon">Opening soon!</span> ${openTimeReceiving.format("LT")} <span data-translation-id="for_receiving">for receiving</span></p>`
```

Comment (experimatt) [reply script.js:146]: I just added `opening_soon`, `for_distribution` and `for_receiving` to the translation spreadsheet, assuming we'll use those identifiers.

File: script.js (hunk 1 above)
Comment (sbeins55) [reply script.js]: Updated!

File: script.js (hunk 2 above)
Comment (sbeins55) [reply script.js]: Updated! Kept the suggested identifiers!

**Comment by sbeins55** [comment]
A couple of things to mention about this PR.

1. When testing with Firefox, we noticed that lodash, moment and list had trouble resolving. This was intermittent and we were able to test functionality was working for the most part.
2. Our group is working on Mac's and we didn't have access to test on Edge.

Thanks to Anna Maxam, Alyson Irizary and Liz Elton for allo working on this together 😄

**Comment by experimatt** [comment]
> 1. When testing with Firefox, we noticed that lodash, moment and list had trouble resolving. This was intermittent and we were able to test functionality was working for the most part.

@sbeins55 You should report this in slack, especially if you encounter the same issues in prod (read: if your changes didn't introduce these problems)

**Comment by alexives** [comment]
>When testing with Firefox, we noticed that lodash, moment and list had trouble resolving. This was intermittent and we were able to test functionality was working for the most part.

That's interesting... I have not had trouble with List loading in firefox, but lodash should get compiled into the app when deployed (it uses webpack to make a single bundle with our dependencies and app code).

**Pull request merged**